zips/zip-0227.rst

```diff
@@ -282,6 +277,20 @@ It contains the following fields:
 The issuance bundle is added within the transaction format as a new bundle. 
 The detailed encoding of the issuance bundle as a part of the V6 transaction format is defined in ZIP 230 [#zip-0230-transaction-format]_.
 
+Computation of ρ
+----------------
+
+We define a function $\mathsf{DeriveIssuedRho} : \mathbb{F}_{q_{\mathbb{P}}} \times \{0 .. 2^{32} - 1\} \times \{0 .. \mathsf{MAX\_ISSUE}\} \to \mathbb{F}_{q_{\mathbb{P}}}$ as follows:
+
+.. math:: \mathsf{DeriveIssuedRho}(\mathsf{nf}, \mathsf{i_{A}}, \mathsf{i_{N}}) := \mathsf{ToBase}^{\mathsf{Orchard}}(\mathsf{PRF}^{\mathsf{expand}}(\mathsf{I2LEOSP}_{256}(\mathsf{nf}), [\mathtt{0x84}] \| \mathsf{I2LEOSP}_{32}(\mathsf{i_{A}}) \| \mathsf{I2LEOSP}_{64}(\mathsf{i_{N}}))) 
```

Comment:
We still need to specify a key to the PRF.

Reply:
also need to say we want to use Blake2s truncated to 254 bits:
![image](https://github.com/user-attachments/assets/c59b2049-957c-4b82-91be-2d51cebd09a3)
https://github.com/zcash/zips/issues/955

Reply:
> both indexes should be encoded the same. (32 bit)

Depends on the outcome of the discussion above (https://github.com/QED-it/zips/pull/90#discussion_r1897329413)

Reply:
> We still need to specify a key to the PRF.

We do, the key is the first argument, aka $\mathsf{nf}$. This follows the usage in ZIP 32, for example: 
<img width="846" alt="image" src="https://github.com/user-attachments/assets/d01ed831-7e44-4258-8d96-d074af8d6c1b" />

It is also accepted notation in the spec: 
<img width="859" alt="image" src="https://github.com/user-attachments/assets/256422cd-87cf-4071-8568-617e06ac4188" />

Reply:
The ref to Blake2s truncated to 254 bits is missing.

Reply:
Regarding BLAKE2s (was looking up details, hence didn't respond until now):
- I understand the use of the $\mathsf{ToBase^{Orchard}} \circ \mathsf{PRF}^{\mathsf{expand}}$ idiom for generating a pseudorandom value over $F_{q_\mathbb{P}}$.
- The alternative suggested is to use BLAKE2s truncated to 254 bits. 
    - From what I could find, I only see BLAKE2s truncated in this way used in Zerocash and Sprout (See [Section 8.6 of the protocol specification](https://zips.z.cash/protocol/protocol.pdf#page=142&zoom=100,72,437)).
    - BLAKE2s (whether or not truncated) does not return a value in $F_{q_\mathbb{P}}$, so we would have to do a $(\mathrm{mod} q_{\mathbb{P}})$ on the output, in which case both truncated and non-truncated versions should be equivalently pseudorandom.
    - I suspect the main benefit (as mentioned by Daira-Emma) is circuit efficiency, possibly due to some details in Halo2? But I haven't figured that out yet. 
    - I also don't know why BLAKE2s instead of BLAKE2b, in that case. It seems the latter is optimized for 64-bit operations while the former is not, I am not sure if that is the reason why.

Reply:
So, based on thoughts in the above comment, we could move from 

$\mathsf{ToBase}^{\mathsf{Orchard}}(\mathsf{PRF}^{\mathsf{expand}}(\mathsf{I2LEOSP}_{256}(\mathsf{nf}), \ldots)$

to 

$\mathsf{BLAKE2s-256}(\mathsf{I2LEOSP}256(\mathsf{nf}) \| \ldots))$ truncated to 254 bits and modulo $q_{\mathbb{P}}$ (github comments aren't letting the math render completely). With the notation like in Section 8.6 linked above.

Reply:
I'd prefer to get a clarification on this before switching to BLAKE2s